### DISCUSSION TITLE
How can I prevent selected tab from being reset when dialog closes?

### DISCUSSION THREAD

**Discussion by ItaiYosephi** [ariakit/ariakit#3942] [Questions] [answered]
I want to prevent `setSelectedId` from being fired when closing the dialog, so the state is preserved next time it opens.
Is this possible?
BTW, it also causes a flicker before closing animation had a chance to finished. 
https://github.com/user-attachments/assets/b368a7ae-6ca7-469e-a13f-631a8db5ab9c
https://stackblitz.com/edit/am32ns-b2vc47?file=dialog-combobox-command-menu%2Findex.tsx

**Comment by diegohaz** [comment]
This looks like a bug. Could you please open a bug report issue?

**Reply by ItaiYosephi** [reply, marked as answer]
opened a bug:
https://github.com/ariakit/ariakit/issues/3963
